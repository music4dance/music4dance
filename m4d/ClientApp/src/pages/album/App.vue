<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import SongTable from "@/components/SongTable.vue";
import AdminFooter from "@/components/AdminFooter.vue";
import { AlbumModel } from "@/models/AlbumModel";
import { TypedJSON } from "typedjson";
import { useSongSelector } from "@/composables/useSongSelector";
import { computed } from "vue";

declare const model_: string;

const model = TypedJSON.parse(model_, AlbumModel)!;

const { songs: selected, select: selectSong } = useSongSelector();

const artistRef = computed(() => `/song/artist?name=${encodeURIComponent(model.artist || "")}`);
const hidden = computed(() => (model.artist ? ["Artist"] : []));
</script>

<template>
  <PageFrame id="app">
    <h1>
      Album: {{ model.title }}
      <span v-if="model.artist"
        >by <a :href="artistRef">{{ model.artist }}</a></span
      >
    </h1>
    <SongTable
      :histories="model.histories!"
      :filter="model.filter"
      :hide-sort="true"
      :hidden-columns="hidden"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
