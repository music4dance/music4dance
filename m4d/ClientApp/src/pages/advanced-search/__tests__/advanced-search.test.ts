import { beforeAll, beforeEach, afterEach, describe, test } from "vitest";
import { testPageSnapshot } from "@/helpers/TestPageSnapshot";
import { mockResizObserver } from "@/helpers/TestHelpers";
import App from "../App.vue";

describe("AdvancedSearch.vue", () => {
  const originalWindowLocation = window.location;

  beforeAll(() => {
    mockResizObserver();
  });

  beforeEach(() => {
    Object.defineProperty(window, "location", {
      configurable: true,
      enumerable: true,
      value: new URL(
        "https://localhost:5001/song/advancedsearchform?filter=v2-Advanced-BOL%2CRMB--Love",
      ),
    });
  });

  afterEach(() => {
    Object.defineProperty(window, "location", {
      configurable: true,
      enumerable: true,
      value: originalWindowLocation,
    });
  });

  test("renders the advanced search page", async () => {
    testPageSnapshot(App);
  });
});
