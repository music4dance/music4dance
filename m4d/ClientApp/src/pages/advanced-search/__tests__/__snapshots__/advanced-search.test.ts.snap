// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdvancedSearch.vue > renders the advanced search page 1`] = `
"<div><span context=\\"[object Object]\\">MainMenu</span>
  <!--v-if-->
  <div id=\\"body-content\\" class=\\"container-fluid body-content\\">
    <div class=\\"mt-2\\"></div>
    <div>
      <h1 class=\\"col-sm-12\\" style=\\"font-size: 22px; text-align: center;\\">Advanced Song Search</h1>
      <div style=\\"max-width: 600px; margin-left: auto; margin-right: auto;\\">
        <form id=\\"advanced-search\\" novalidate=\\"\\" class=\\"\\">
          <div class=\\"mb-3\\" id=\\"search-string-group\\" role=\\"group\\"><label id=\\"__BVID__0001___BV__BV_label___\\" for=\\"search-string\\" class=\\"form-label d-block\\">Keywords:</label>
            <div class=\\"\\">
              <div class=\\"input-group mb-1\\" role=\\"group\\">
                <!----><input id=\\"search-everywhere\\" class=\\"form-control\\" type=\\"text\\" placeholder=\\"Enter artist, title, etc. OR a Spotify or Apple share link\\">
                <!---->
              </div><button class=\\"btn btn-sm btn-outline-secondary rounded-0 float-end\\" type=\\"button\\"><svg viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                  <path fill=\\"currentColor\\" d=\\"M3.204 5h9.592L8 10.481zm-.753.659l4.796 5.48a1 1 0 0 0 1.506 0l4.796-5.48c.566-.647.106-1.659-.753-1.659H3.204a1 1 0 0 0-.753 1.659\\"></path>
                </svg> more <svg viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                  <path fill=\\"currentColor\\" d=\\"M3.204 5h9.592L8 10.481zm-.753.659l4.796 5.48a1 1 0 0 0 1.506 0l4.796-5.48c.566-.647.106-1.659-.753-1.659H3.204a1 1 0 0 0-.753 1.659\\"></path>
                </svg></button>
              <!---->
              <!---->
              <!---->
            </div>
          </div>
          <fieldset class=\\"\\" id=\\"dance-group\\">
            <legend id=\\"__BVID__0002___BV__BV_label___\\" tabindex=\\"-1\\" class=\\"form-label bv-no-focus-ring col-form-label pt-0\\">Dances:</legend>
            <div class=\\"\\">
              <div style=\\"border: 1px solid #ced4da;\\">
                <div id=\\"dance-selector\\" class=\\"b-form-tags form-control h-auto mb-2 mt-2\\" role=\\"group\\" tabindex=\\"-1\\" style=\\"border-style: hidden;\\"><output id=\\"__BVID__0003___BV___selected_tags__\\" class=\\"visually-hidden\\" for=\\"__BVID__0003___BV___input__\\" aria-live=\\"off\\" aria-atomic=\\"true\\" aria-relevant=\\"additions text\\">BOL, RMB</output>
                  <div id=\\"__BVID__0003___BV___removed_tags__\\" role=\\"status\\" aria-live=\\"off\\" aria-atomic=\\"true\\" class=\\"visually-hidden\\"> (Tag removed) </div>
                  <ul class=\\"list-inline d-inline-block mb-2\\">
                    <li class=\\"list-inline-item\\"><span id=\\"__BVID__0004___BV___\\" title=\\"Bolero\\" class=\\"badge b-form-tag d-inline-flex align-items-center mw-100 text-bg-primary\\" aria-labelledby=\\"__BVID__0004___BV___taglabel__\\" data-bs-theme=\\"dark\\"><span id=\\"__BVID__0004___BV___taglabel__\\" class=\\"b-form-tag-content flex-grow-1 text-truncate\\">Bolero</span><button type=\\"button\\" class=\\"btn-close b-form-tag-remove\\" aria-label=\\"Remove tag\\" aria-keyshortcuts=\\"Delete\\" aria-describedby=\\"__BVID__0004___BV___taglabel__\\"></button></span></li>
                    <li class=\\"list-inline-item\\"><span id=\\"__BVID__0005___BV___\\" title=\\"Rumba\\" class=\\"badge b-form-tag d-inline-flex align-items-center mw-100 text-bg-primary\\" aria-labelledby=\\"__BVID__0005___BV___taglabel__\\" data-bs-theme=\\"dark\\"><span id=\\"__BVID__0005___BV___taglabel__\\" class=\\"b-form-tag-content flex-grow-1 text-truncate\\">Rumba</span><button type=\\"button\\" class=\\"btn-close b-form-tag-remove\\" aria-label=\\"Remove tag\\" aria-keyshortcuts=\\"Delete\\" aria-describedby=\\"__BVID__0005___BV___taglabel__\\"></button></span></li>
                  </ul>
                  <div class=\\"btn-group tag-dropdown\\" block=\\"\\"><button class=\\"btn btn-sm btn-outline-secondary dropdown-toggle\\" type=\\"button\\" id=\\"__BVID__0006___BV_dropdown__\\" aria-expanded=\\"false\\" aria-haspopup=\\"menu\\"><svg viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                        <path fill=\\"currentColor\\" d=\\"M2 1a1 1 0 0 0-1 1v4.586a1 1 0 0 0 .293.707l7 7a1 1 0 0 0 1.414 0l4.586-4.586a1 1 0 0 0 0-1.414l-7-7A1 1 0 0 0 6.586 1zm4 3.5a1.5 1.5 0 1 1-3 0a1.5 1.5 0 0 1 3 0\\"></path>
                      </svg> Choose Dances</button>
                    <!---->
                    <!--teleport start-->
                    <ul style=\\"position: absolute; left: 0px; top: 0px; display: none;\\" class=\\"dropdown-menu overflow-auto\\" aria-labelledby=\\"__BVID__0006___BV_dropdown__\\" role=\\"menu\\">
                      <div>
                        <li role=\\"presentation\\">
                          <form class=\\"dropdown-item-text\\">
                            <div class=\\"row d-flex flex-wrap mb-0\\" id=\\"__BVID__0008___BV___\\" role=\\"group\\"><label class=\\"col-md-auto col-form-label col-form-label col-form-label-sm\\" id=\\"__BVID__0007___BV__BV_label___\\" for=\\"tag-search-input\\">Search Dances</label>
                              <div class=\\"col\\"><input id=\\"__BVID__0009___BV_input__\\" class=\\"form-control form-control-sm\\" type=\\"search\\" autocomplete=\\"off\\">
                                <!---->
                                <!---->
                                <!---->
                              </div>
                            </div>
                          </form>
                        </li>
                        <li role=\\"presentation\\">
                          <hr class=\\"dropdown-divider\\" role=\\"separator\\" aria-orientation=\\"horizontal\\">
                        </li>
                        <!--v-if-->
                        <div>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Argentine Tango</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Balboa</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Bachata</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Broadway</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Ballet</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Blues</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Bossa Nova</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Bollywood</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Country Two Step</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Castle Foxtrot</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Cha Cha</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Collegiate Shag</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Cumbia</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Contemporary</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Carolina Shag</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Charleston</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Cross-step Waltz</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Disco</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">East Coast Swing</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Freestyle</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Foxtrot</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Hip-Hop</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Hustle</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Jazz</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Jive</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Jump Swing</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Lindy Hop</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Latin</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Mambo</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Milonga</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Merengue</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Other</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Motown</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Night Club Two Step</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Neo Tango</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Peabody</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Paso Doble</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Polka</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Performance</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Quickstep</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Slow Foxtrot</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Salsa</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Samba</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Single Swing</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Swing</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Slow Waltz</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Tap</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Tango (Ballroom)</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Tango Vals</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Tango</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Viennese Waltz</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">West Coast Swing</button></li>
                          <li role=\\"presentation\\"><button role=\\"menu\\" type=\\"button\\" class=\\"dropdown-item\\">Waltz</button></li>
                        </div>
                      </div>
                    </ul>
                    <!--teleport end-->
                  </div>
                  <!---->
                </div>
                <div class=\\"d-flex justify-content-between w-100 mx-1 mb-2\\">
                  <div id=\\"dance-connector\\" role=\\"radiogroup\\" class=\\"btn-group btn-group-sm bv-no-focus-ring mx-3\\" tabindex=\\"-1\\"><input id=\\"__BVID__0010___BV_form-check__\\" class=\\"btn-check\\" type=\\"radio\\" name=\\"__BVID__0011___BV_checkbox__\\" value=\\"any\\"><label for=\\"__BVID__0010___BV_form-check__\\" class=\\"btn btn-outline-secondary btn-sm\\">Any</label><input id=\\"__BVID__0012___BV_form-check__\\" class=\\"btn-check\\" type=\\"radio\\" name=\\"__BVID__0011___BV_checkbox__\\" value=\\"all\\"><label for=\\"__BVID__0012___BV_form-check__\\" class=\\"btn btn-outline-secondary btn-sm\\">All</label></div>
                </div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
          </fieldset>
          <fieldset class=\\"\\" id=\\"include-tags-group\\">
            <legend id=\\"__BVID__0013___BV__BV_label___\\" tabindex=\\"-1\\" class=\\"form-label bv-no-focus-ring col-form-label pt-0\\">Include Tags:</legend>
            <div class=\\"\\">
              <div id=\\"include-tags\\" class=\\"b-form-tags form-control h-auto mb-2\\" role=\\"group\\" tabindex=\\"-1\\" style=\\"border-style: hidden;\\"><output id=\\"__BVID__0014___BV___selected_tags__\\" class=\\"visually-hidden\\" for=\\"__BVID__0014___BV___input__\\" aria-live=\\"off\\" aria-atomic=\\"true\\" aria-relevant=\\"additions text\\"></output>
                <div id=\\"__BVID__0014___BV___removed_tags__\\" role=\\"status\\" aria-live=\\"off\\" aria-atomic=\\"true\\" class=\\"visually-hidden\\"> (Tag removed) </div>
                <!--v-if-->
                <div class=\\"btn-group tag-dropdown\\" block=\\"\\"><button class=\\"btn btn-sm btn-outline-secondary dropdown-toggle\\" type=\\"button\\" id=\\"__BVID__0015___BV_dropdown__\\" aria-expanded=\\"false\\" aria-haspopup=\\"menu\\"><svg viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                      <path fill=\\"currentColor\\" d=\\"M2 1a1 1 0 0 0-1 1v4.586a1 1 0 0 0 .293.707l7 7a1 1 0 0 0 1.414 0l4.586-4.586a1 1 0 0 0 0-1.414l-7-7A1 1 0 0 0 6.586 1zm4 3.5a1.5 1.5 0 1 1-3 0a1.5 1.5 0 0 1 3 0\\"></path>
                    </svg> Choose Tags to Include</button>
                  <!---->
                  <!--teleport start-->
                  <ul style=\\"position: absolute; left: 0px; top: 0px; display: none;\\" class=\\"dropdown-menu overflow-auto\\" aria-labelledby=\\"__BVID__0015___BV_dropdown__\\" role=\\"menu\\">
                    <div>
                      <li role=\\"presentation\\">
                        <form class=\\"dropdown-item-text\\">
                          <div class=\\"row d-flex flex-wrap mb-0\\" id=\\"__BVID__0017___BV___\\" role=\\"group\\"><label class=\\"col-md-auto col-form-label col-form-label col-form-label-sm\\" id=\\"__BVID__0016___BV__BV_label___\\" for=\\"tag-search-input\\">Search Tags</label>
                            <div class=\\"col\\"><input id=\\"__BVID__0018___BV_input__\\" class=\\"form-control form-control-sm\\" type=\\"search\\" autocomplete=\\"off\\">
                              <!---->
                              <!---->
                              <!---->
                            </div>
                          </div>
                        </form>
                      </li>
                      <li role=\\"presentation\\">
                        <hr class=\\"dropdown-divider\\" role=\\"separator\\" aria-orientation=\\"horizontal\\">
                      </li>
                      <!--v-if-->
                      <li role=\\"presentation\\"><span class=\\"dropdown-item-text\\"> Start typing to see a list of matching tags </span></li>
                    </div>
                  </ul>
                  <!--teleport end-->
                </div>
                <!---->
              </div>
              <!---->
              <!---->
              <!---->
            </div>
          </fieldset>
          <fieldset class=\\"\\" id=\\"exclude-tags-group\\">
            <legend id=\\"__BVID__0019___BV__BV_label___\\" tabindex=\\"-1\\" class=\\"form-label bv-no-focus-ring col-form-label pt-0\\">Exclude Tags:</legend>
            <div class=\\"\\">
              <div id=\\"exclude-tags\\" class=\\"b-form-tags form-control h-auto mb-2\\" role=\\"group\\" tabindex=\\"-1\\" style=\\"border-style: hidden;\\"><output id=\\"__BVID__0020___BV___selected_tags__\\" class=\\"visually-hidden\\" for=\\"__BVID__0020___BV___input__\\" aria-live=\\"off\\" aria-atomic=\\"true\\" aria-relevant=\\"additions text\\"></output>
                <div id=\\"__BVID__0020___BV___removed_tags__\\" role=\\"status\\" aria-live=\\"off\\" aria-atomic=\\"true\\" class=\\"visually-hidden\\"> (Tag removed) </div>
                <!--v-if-->
                <div class=\\"btn-group tag-dropdown\\" block=\\"\\"><button class=\\"btn btn-sm btn-outline-secondary dropdown-toggle\\" type=\\"button\\" id=\\"__BVID__0021___BV_dropdown__\\" aria-expanded=\\"false\\" aria-haspopup=\\"menu\\"><svg viewBox=\\"0 0 16 16\\" width=\\"1.2em\\" height=\\"1.2em\\">
                      <path fill=\\"currentColor\\" d=\\"M2 1a1 1 0 0 0-1 1v4.586a1 1 0 0 0 .293.707l7 7a1 1 0 0 0 1.414 0l4.586-4.586a1 1 0 0 0 0-1.414l-7-7A1 1 0 0 0 6.586 1zm4 3.5a1.5 1.5 0 1 1-3 0a1.5 1.5 0 0 1 3 0\\"></path>
                    </svg> Choose Tags to Exclude</button>
                  <!---->
                  <!--teleport start-->
                  <ul style=\\"position: absolute; left: 0px; top: 0px; display: none;\\" class=\\"dropdown-menu overflow-auto\\" aria-labelledby=\\"__BVID__0021___BV_dropdown__\\" role=\\"menu\\">
                    <div>
                      <li role=\\"presentation\\">
                        <form class=\\"dropdown-item-text\\">
                          <div class=\\"row d-flex flex-wrap mb-0\\" id=\\"__BVID__0023___BV___\\" role=\\"group\\"><label class=\\"col-md-auto col-form-label col-form-label col-form-label-sm\\" id=\\"__BVID__0022___BV__BV_label___\\" for=\\"tag-search-input\\">Search Tags</label>
                            <div class=\\"col\\"><input id=\\"__BVID__0024___BV_input__\\" class=\\"form-control form-control-sm\\" type=\\"search\\" autocomplete=\\"off\\">
                              <!---->
                              <!---->
                              <!---->
                            </div>
                          </div>
                        </form>
                      </li>
                      <li role=\\"presentation\\">
                        <hr class=\\"dropdown-divider\\" role=\\"separator\\" aria-orientation=\\"horizontal\\">
                      </li>
                      <!--v-if-->
                      <li role=\\"presentation\\"><span class=\\"dropdown-item-text\\"> Start typing to see a list of matching tags </span></li>
                    </div>
                  </ul>
                  <!--teleport end-->
                </div>
                <!---->
              </div>
              <!---->
              <!---->
              <!---->
            </div>
          </fieldset>
          <div class=\\"\\" id=\\"tempo-range-group\\" role=\\"group\\"><label id=\\"__BVID__0025___BV__BV_label___\\" for=\\"tempo-range\\" class=\\"form-label d-block\\">Tempo range (BPM):</label>
            <div class=\\"\\">
              <fieldset class=\\"\\" id=\\"tempo-range\\">
                <!---->
                <div class=\\"\\">
                  <div class=\\"d-flex\\"><input id=\\"tempo-min\\" class=\\"form-control\\" type=\\"number\\" min=\\"0\\" max=\\"400\\" style=\\"width: 6rem;\\"><span class=\\"mx-2 pt-2\\">to</span><input id=\\"tempo-max\\" class=\\"form-control\\" type=\\"number\\" min=\\"0\\" max=\\"400\\" style=\\"width: 6rem;\\">
                    <div class=\\"invalid-feedback\\">Tempos must be between 0 and 400 BPM</div>
                  </div>
                  <!---->
                  <!---->
                  <!---->
                </div>
              </fieldset>
              <!---->
              <!---->
              <!---->
            </div>
          </div>
          <div class=\\"\\" id=\\"length-range-group\\" role=\\"group\\"><label id=\\"__BVID__0026___BV__BV_label___\\" for=\\"length-range\\" class=\\"form-label d-block\\">Length range (seconds):</label>
            <div class=\\"\\">
              <fieldset class=\\"\\" id=\\"length-range\\">
                <!---->
                <div class=\\"\\">
                  <div class=\\"d-flex\\"><input id=\\"length-min\\" class=\\"form-control\\" type=\\"number\\" min=\\"0\\" max=\\"600\\" style=\\"width: 6rem;\\"><span class=\\"mx-2 pt-2\\">to</span><input id=\\"length-max\\" class=\\"form-control\\" type=\\"number\\" min=\\"0\\" max=\\"600\\" style=\\"width: 6rem;\\">
                    <div class=\\"invalid-feedback\\">Length must be between 0 and 600 seconds</div>
                  </div>
                  <!---->
                  <!---->
                  <!---->
                </div>
              </fieldset>
              <!---->
              <!---->
              <!---->
            </div>
          </div>
          <div class=\\"\\" id=\\"activity-group\\" role=\\"group\\"><label id=\\"__BVID__0027___BV__BV_label___\\" for=\\"activity\\" class=\\"form-label d-block\\">By User:</label>
            <div class=\\"\\">
              <div class=\\"d-flex\\"><input id=\\"user\\" class=\\"form-control mr-2\\" type=\\"text\\" placeholder=\\"UserName or me\\" style=\\"width: 10rem;\\"><select id=\\"activity\\" class=\\"form-select\\">
                  <option value=\\"NT\\">Don't filter on user activity</option>
                </select></div>
              <!---->
              <!---->
              <!---->
            </div>
          </div>
          <div class=\\"mx-2 mb-2\\" id=\\"services-group\\" role=\\"group\\"><label id=\\"__BVID__0028___BV__BV_label___\\" for=\\"services\\" class=\\"form-label d-block\\">Available on:</label>
            <div class=\\"\\">
              <div id=\\"services\\" role=\\"group\\" class=\\"btn-group-md bv-no-focus-ring\\" tabindex=\\"-1\\">
                <div class=\\"form-check form-check-inline\\"><input id=\\"__BVID__0029___BV_form-check__\\" class=\\"form-check-input\\" type=\\"checkbox\\" name=\\"__BVID__0030___BV_checkbox__\\" true-value=\\"A\\" false-value=\\"false\\" value=\\"A\\"><label for=\\"__BVID__0029___BV_form-check__\\" class=\\"form-check-label\\">Amazon</label></div>
                <div class=\\"form-check form-check-inline\\"><input id=\\"__BVID__0031___BV_form-check__\\" class=\\"form-check-input\\" type=\\"checkbox\\" name=\\"__BVID__0030___BV_checkbox__\\" true-value=\\"I\\" false-value=\\"false\\" value=\\"I\\"><label for=\\"__BVID__0031___BV_form-check__\\" class=\\"form-check-label\\">ITunes</label></div>
                <div class=\\"form-check form-check-inline\\"><input id=\\"__BVID__0032___BV_form-check__\\" class=\\"form-check-input\\" type=\\"checkbox\\" name=\\"__BVID__0030___BV_checkbox__\\" true-value=\\"S\\" false-value=\\"false\\" value=\\"S\\"><label for=\\"__BVID__0032___BV_form-check__\\" class=\\"form-check-label\\">Spotify</label></div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
          </div>
          <div class=\\"mx-2 mb-2\\" id=\\"bonuses-group\\" role=\\"group\\"><label id=\\"__BVID__0033___BV__BV_label___\\" for=\\"bonuses\\" class=\\"form-label d-block\\">Bonus content:</label>
            <div class=\\"\\">
              <div id=\\"bonuses\\" role=\\"group\\" class=\\"btn-group-md bv-no-focus-ring\\" tabindex=\\"-1\\">
                <div class=\\"form-check form-check-inline\\"><input id=\\"__BVID__0035___BV_form-check__\\" class=\\"form-check-input\\" type=\\"checkbox\\" name=\\"__BVID__0036___BV_checkbox__\\" true-value=\\"P\\" false-value=\\"false\\" value=\\"P\\"><label for=\\"__BVID__0035___BV_form-check__\\" class=\\"form-check-label\\">Not found in any publisher catalog</label></div>
                <div class=\\"form-check form-check-inline\\"><input id=\\"__BVID__0037___BV_form-check__\\" class=\\"form-check-input\\" type=\\"checkbox\\" name=\\"__BVID__0036___BV_checkbox__\\" true-value=\\"D\\" false-value=\\"false\\" value=\\"D\\"><label for=\\"__BVID__0037___BV_form-check__\\" class=\\"form-check-label\\">Not categorized by dance</label></div>
              </div>
              <!---->
              <!----><small id=\\"__BVID__0034___BV__BV_description___\\" class=\\"text-body-secondary form-text\\"><a href=\\"https://music4dance.blog/music4dance-help/subscriptions/\\">Premium content</a></small>
            </div>
          </div>
          <div class=\\"\\" id=\\"sort-group\\" role=\\"group\\"><label id=\\"__BVID__0038___BV__BV_label___\\" for=\\"sort\\" class=\\"form-label d-block\\">Sort By:</label>
            <div class=\\"\\"><select id=\\"sort\\" class=\\"form-select\\" required=\\"\\" aria-required=\\"true\\">
                <option value=\\"Title\\">Title</option>
                <option value=\\"Artist\\">Artist</option>
                <option value=\\"Tempo\\">Tempo</option>
                <option value=\\"Length\\">Length</option>
                <option value=\\"Modified\\">Last Modified</option>
                <option value=\\"Edited\\">Last Edited</option>
                <option value=\\"Created\\">When Added</option>
                <option value=\\"Energy\\">Energy</option>
                <option value=\\"Mood\\">Mood</option>
                <option value=\\"Beat\\">Strength of Beat</option>
                <option value=\\"Comments\\">Comments</option>
                <option>Closest Match</option>
              </select>
              <div id=\\"sort-order\\" role=\\"radiogroup\\" class=\\"btn-group-md bv-no-focus-ring mt-2\\" tabindex=\\"-1\\">
                <div class=\\"form-check form-check-inline\\"><input id=\\"__BVID__0039___BV_form-check__\\" class=\\"form-check-input\\" type=\\"radio\\" name=\\"sort-order\\" value=\\"asc\\"><label for=\\"__BVID__0039___BV_form-check__\\" class=\\"form-check-label\\">Ascending (A-Z, Slow-Fast, Newest-Oldest, Shortest-Longest)</label></div>
                <div class=\\"form-check form-check-inline\\"><input id=\\"__BVID__0040___BV_form-check__\\" class=\\"form-check-input\\" type=\\"radio\\" name=\\"sort-order\\" value=\\"desc\\"><label for=\\"__BVID__0040___BV_form-check__\\" class=\\"form-check-label\\">Descending (Z-A, Fast-Slow, Oldest-Newest, Longest-Shortest)</label></div>
              </div>
              <!---->
              <!---->
              <!---->
            </div>
          </div>
          <div class=\\"d-flex justify-content-between w-100 mx-1 mb-2\\"><button class=\\"btn btn-md btn-secondary\\" type=\\"reset\\">Reset</button><button class=\\"btn btn-md btn-primary\\" type=\\"submit\\">Submit</button></div>
        </form>
        <!--v-if-->
      </div>
    </div>
  </div>
  <div id=\\"footer-content\\">
    <hr>
    <footer>
      <p> © 2024 - <a href=\\"https://www.music4dance.net\\">Music4Dance.net</a> - <a href=\\"https://www.music4dance.net/home/sitemap\\">Site Map</a> - <a href=\\"https://www.music4dance.net/home/termsofservice\\">Terms of Service</a> - <a href=\\"https://www.music4dance.net/home/privacypolicy\\">Privacy Policy</a> - <a href=\\"https://www.music4dance.net/home/credits\\">Credits</a> - <a>Help</a></p>
    </footer>
  </div>
</div>"
`;
